packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -225,51 +225,50 @@ class SlideTransition extends AnimatedWidget {
   }
 }
 
-/// Animates the scale of a transformed widget.
+/// Signature for the callback to [MatrixTransition.onTransform].
 ///
-/// Here's an illustration of the [ScaleTransition] widget, with it's [alignment]
-/// animated by a [CurvedAnimation] set to [Curves.fastOutSlowIn]:
-/// {@animation 300 378 https://flutter.github.io/assets-for-api-docs/assets/widgets/scale_transition.mp4}
-///
-/// {@tool dartpad}
-/// The following code implements the [ScaleTransition] as seen in the video
-/// above:
+/// Computes a [Matrix4] to be used in the [MatrixTransition] transformed widget
+/// from the [MatrixTransition.animation] value.
+typedef TransformCallback = Matrix4 Function(double);
+
+/// Animates the [Matrix4] of a transformed widget.
 ///
-/// ** See code in examples/api/lib/widgets/transitions/scale_transition.0.dart **
-/// {@end-tool}
+/// The [onTransform] callback computes a [Matrix4] from the animated value.
 ///
 /// See also:
 ///
-///  * [PositionedTransition], a widget that animates its child from a start
-///    position to an end position over the lifetime of the animation.
-///  * [RelativePositionedTransition], a widget that transitions its child's
-///    position based on the value of a rectangle relative to a bounding box.
-///  * [SizeTransition], a widget that animates its own size and clips and
-///    aligns its child.
-class ScaleTransition extends AnimatedWidget {
-  /// Creates a scale transition.
+///  * [ScaleTransition], which animates the scale of a widget, by providing a
+///    matrix which scales along the X and Y axis.
+///  * [RotationTransition], which animates the rotation of a widget, by
+///    providing a matrix which rotates along the Z axis.
+class MatrixTransition extends AnimatedWidget {
+  /// Creates a matrix transition.
   ///
-  /// The [scale] argument must not be null. The [alignment] argument defaults
-  /// to [Alignment.center].
-  const ScaleTransition({
+  /// The [animation] and [onTransform] arguments must not be null.
```

Comment:
nit: can you remove this sentence while you're in here? With null safety fully rolled out there is no need to document this explicitly anymore.